homeassistant/components/thermostat/zwave.py

```diff
@@ -0,0 +1,108 @@
+"""
+ZWave Thermostat.
+
+"""
+
+# Because we do not compile openzwave on CI
+# pylint: disable=import-error
+from homeassistant.components.thermostat import DOMAIN
+from homeassistant.components.thermostat import ThermostatDevice
+from homeassistant.components.zwave import (
+    ATTR_NODE_ID, ATTR_VALUE_ID, NETWORK,
+    ZWaveDeviceEntity)
+from homeassistant.const import TEMP_FAHRENHEIT
+
+CONF_NAME = 'name'
+DEFAULT_NAME = 'ZWave Thermostat'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the ZWave thermostats."""
+    if discovery_info is None or NETWORK is None:
+        return
+
+    node = NETWORK.nodes[discovery_info[ATTR_NODE_ID]]
+    value = node.values[discovery_info[ATTR_VALUE_ID]]
+    value.set_change_verified(False)
+    add_devices([ZWaveThermostat(value)])
+
+
+# pylint: disable=too-many-arguments
+class ZWaveThermostat(ZWaveDeviceEntity, ThermostatDevice):
+    """Represents a HeatControl thermostat."""
+    def __init__(self, value):
+        from openzwave.network import ZWaveNetwork
+        from pydispatch import dispatcher
+        from homeassistant.helpers.temperature import convert
+        ZWaveDeviceEntity.__init__(self, value, DOMAIN)
+        self._node = value.node
+        self._target_temperature = round(convert(71,
+                                                 TEMP_FAHRENHEIT,
+                                                 self.hass.config.temperature_unit))
+        self._current_temperature = round(convert(90,
+                                                  TEMP_FAHRENHEIT,
+                                                  self.hass.config.temperature_unit))
+        self._current_operation = STATE_IDLE
+        self._current_operation_state = STATE_IDLE
+        self.update_properties()
+        # register listener
+        dispatcher.connect(
+            self.value_changed, ZWaveNetwork.SIGNAL_VALUE_CHANGED)
+
+    def value_changed(self, value):
+        """Called when a value has changed on the network."""
+        if self._value.node == value.node:
+            self.update_properties()
+            self.update_ha_state()
+
+    def update_properties(self):
+        """ Callback on data change for the registered node/value pair. """
+        # set point
+        for _, value in self._node.get_values_for_command_class(0x43).items():
+            if int(value.data) != 0:
+                self._target_temperature = value.data
+        # Operation
+        for _, value in self._node.get_values_for_command_class(0x40).items():
+            self._current_operation = value.data
+        # Current Temp
+        for _, value in self._node.get_values_for_command_class(0x31).items():
+            if int(value.data) != 0:
+                self._current_temperature = value.data
```

Comment:
Here you can get the unit of measurement too from `value.unit` to input in the `unit_of_measurement(self)`